js/base/Exchange.js

```diff
@@ -1682,6 +1682,195 @@ module.exports = class Exchange {
         return this.filterByArray (result, 'symbol', symbols, false);
     }
 
+    safePosition (position, market = undefined) {
+        market = this.market (position['symbol']);
+        const side = position['side'];
+        const realizedPnlString = this.safeString (position, 'realizedPnl');
+        const entryPriceString = this.safeString (position, 'entryPrice');
+        const markPriceString = this.safeString (position, 'markPrice');
+        const lastPriceString = this.safeString (position, 'lastPrice');
+        // support deprecated 'marginType'
+        const marginMode = this.safeString2 (position, 'marginMode', 'marginType');
+        //
+        // CONTRACT SIZE
+        //
+        let contractSizeString = this.safeString (position, 'contractSize');
+        if (contractSizeString === undefined) {
+            contractSizeString = this.safeString (market, 'contractSize');
+        }
```

Comment:
```suggestion
        const marketContractSize = this.safeString (market, 'contractSize')
        const contractSizeString = this.safeString (position, 'contractSize', marketContractSize);
```

Reply:
for the case of simplicity of the PR/edits, I prefer not to touch other things, even simple edits. let's leave them as they are, and after merging I'll re-do the optimization of the codes.

Reply:
one slight note - the above suggestion ( i missed it) was related to #12765 , so you can comment there.
second slight note - i think there is tiniest performance gain if we don't access/parse `market` object, if the contract-size was already defined, thus ` === undefined` here is better imho.